File: Serene/Serene.Script/Administration/Language/LanguageDialog.cs

namespace Serene.Administration
{
    using Serenity;
    using System.Runtime.CompilerServices;

    [Imported]
    public class LanguageDialog : EntityDialog<LanguageRow>
    {
    }

    /* 
    This class has been ported to TypeScript. See LanguageDialog.ts
    Code below is only a reference for those who want to use Saltaralle

    [IdProperty("Id"), NameProperty("LanguageName")]
    [FormKey("Administration.Language"), LocalTextPrefix("Administration.Language"), Service("Administration/Language")]
    public class LanguageDialog : EntityDialog<LanguageRow>, IAsyncInit
    {
    }
    */
}

File: Serene/Serene.Web/Modules/Administration/Language/LanguageDialog.ts
namespace Serene.Administration {

    @Serenity.Decorators.registerClass()
    @Serenity.Decorators.responsive()
    export class LanguageDialog extends Serenity.EntityDialog<LanguageRow, any> {
        protected getFormKey() { return LanguageForm.formKey; }
        protected getIdProperty() { return LanguageRow.idProperty; }
        protected getLocalTextPrefix() { return LanguageRow.localTextPrefix; }
        protected getNameProperty() { return LanguageRow.nameProperty; }
        protected getService() { return LanguageService.baseUrl; }

        protected form = new LanguageForm(this.idPrefix);
    }
}